src/components/CommentAdder.jsx

```diff
@@ -0,0 +1,60 @@
+import { useState } from "react";
+import { useParams } from "react-router-dom";
+import { postComment } from "../utils/api";
+
+export default function CommentAdder(props) {
+  const { article_id } = useParams();
+  const { allComments, setAllComments } = props;
+
+  const [body, setBody] = useState("");
+  const [error, setError] = useState(null);
+  const [successResponse, setSuccessReponse] = useState(null);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const handleChange = (event) => {
+    setBody(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    setIsSubmitting(true);
+
+    postComment(article_id, body)
+      .then((comment) => {
+        setAllComments([comment, ...allComments]);
+        setBody("");
+        setError(null);
+        setSuccessReponse("Comment posted successfully!");
+        setTimeout(() => {
+          setSuccessReponse(null);
+        }, 3000);
```

Comment:
When you finish the project and come back to it you might like to look at toast notifications in react